### PR TITLE
Summary

### PR DESCRIPTION
**Pull request by this-username-is-taken**
ios simulator screen shot feb 5 2014 8 36 08 pm

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -11,3 +11,4 @@ config/*
 config/**/*
 !settings.py
 !*.sample.py
+static/store_imgs/*
```
Comment (gregwym) [inline .gitignore]: always make sure there is new line at EOF, or config your editor to do it for you.

**Comment by gregwym** [comment]
1. Edit your PR title to express what its about, be specific. 
2. pull the latest `dev` branch and rebase yours on that, otherwise it cannot be auto merged. (You don't have to do this now, but doing it now will make the diff easier to look at. )

File: .gitignore (hunk 1 above)
Comment (gregwym) [inline .gitignore]: Thanks for fixing it!

File: ntemplates/delivery/detail.html [hunk 2]
```diff
@@ -1,79 +1,51 @@
+<meta name="viewport" content="width=device-width">
```
Comment (gregwym) [inline ntemplates/delivery/detail.html]: I don't think this line will be shown on the page. You should move this line into `/ntemplate/delivery/base.html` within the `extra_script` block. So it will show inside the `<head>` tag. And applies to all the delivery page, which I guess is what you want.

File: ntemplates/delivery/detail.html [hunk 3]
```diff
@@ -1,79 +1,51 @@
...
+
 {% extends "delivery/base.html" %}
+
+
```
Comment (gregwym) [inline ntemplates/delivery/detail.html:4]: Avoid unnecessary changes, otherwise they may result in unexpected conflict when merging.

File: ntemplates/delivery/detail.html (hunk 2 above)
Comment (gregwym) [inline ntemplates/delivery/detail.html]: Or `page_css` block. up to you.

File: ntemplates/delivery/detail.html [hunk 4]
```diff
@@ -1,79 +1,51 @@
...
 
 {% localtime on %}
 
-{% block content %}
 
-<h1>Delivery Window</h1>
-<h2>{{ delivery_window.store }}: {{ delivery_window }}</h2>
+{% block extra_script %}
+<script type="text/javascript" src="{% static "javascript/delivery.js" %}"></script>
+{% endblock %}
```
Comment (gregwym) [inline ntemplates/delivery/detail.html:13]: Same thing for this block. I guess this script will be applicable for all delivery pages. So its better to be in the `delivery/base.html` file.

File: ntemplates/delivery/detail.html [hunk 5]
```diff
@@ -1,79 +1,51 @@
...
+    {% if item.on_sale %}
+    <span> Sales Price: {{ item.sales_price }}</span>
+    {% else %}
+    <span> Price: {{ item.price }}</span>
+    {% endif %}
+    {% if item.category.shop_location_note %}
+    <br>
+    <span> Location: {{ item.category.shop_location_note }}</span>
+    {% endif %}
+  </div>
```
Comment (gregwym) [inline ntemplates/delivery/detail.html]: Why this `div` block was indented? It is at the same level as the `h3`.

File: ntemplates/delivery/detail.html [hunk 6]
```diff
@@ -1,79 +1,51 @@
...
+  {% endfor %}
+
 <ul>
```
Comment (gregwym) [inline ntemplates/delivery/detail.html:37]: We may want a title here to indicate this is a new section.

File: ntemplates/delivery/detail.html [hunk 7]
```diff
@@ -1,79 +1,51 @@
...
-  {% for order in delivery_window.orders.all %}
+  {% for order in delivery_window.waiting_orders %}
   <li>
-    <div>
+    <div class="order_title">
       <a href="{% url 'order:show' order.invoice.invoice_num %}">
-        Order {{ order }}: {{ order.get_status_display }} ({{ order.items.count }} items)
+        Order {{ order }}: {{ order.invoice.customer_name }} ({{ order.items.count }} items)
```
Comment (gregwym) [inline ntemplates/delivery/detail.html:42]: As I said in the last PR, the `status` is still important to display.

File: ntemplates/delivery/detail.html [hunk 8]
```diff
@@ -1,79 +1,51 @@
...
-          <td>
-            Email:
-          </td>
-          <td>
-            {{ order.invoice.email }}
-          </td>
-        </tr>
-      </tbody>
-      </table>
-    </div>
```
Comment (gregwym) [inline ntemplates/delivery/detail.html:72]: Why this table is removed? Please explain all your design decisions in the PR description. Or even better, we should discus this prior to making changes. So save both of us time.

File: ntemplates/delivery/detail.html [hunk 9]
```diff
@@ -86,27 +58,7 @@
...
-    {% if item.category.shop_location_note %}
-    <br>
-    <span> Location: {{ item.category.shop_location_note }}</span>
-    {% endif %}
-  </li>
-  {% endfor %}
-</ol>
-
-<a href="{% url 'delivery:summary' %}">Go back to summary</a>
+<h3><a href="{% url 'delivery:summary' %}">Go back to summary</a></h3>
```
Comment (gregwym) [inline ntemplates/delivery/detail.html]: This is not a title/header, shouldn't be in a header tag.

File: ntemplates/delivery/summary.html [hunk 10]
```diff
@@ -1,3 +1,5 @@
+<meta name="viewport" content="width=device-width">
+
```
Comment (gregwym) [inline ntemplates/delivery/summary.html]: As I wrote above. move this to `delivery/base.html`.

File: ntemplates/delivery/detail.html (hunk 2 above)
Comment (gregwym) [inline ntemplates/delivery/detail.html]: Or you can add a `head_extra` block in the `/common/base.html` which make it even better structured. Cuz we don't really have a proper block available.

File: ntemplates/delivery/detail.html (hunk 4 above)
Comment (this-username-is-taken) [inline ntemplates/delivery/detail.html:13]: This is only used in the detail page, I will change the other ones.

File: ntemplates/delivery/detail.html (hunk 5 above)
Comment (this-username-is-taken) [inline ntemplates/delivery/detail.html]: good catch. I just followed the for loop indentation. I will un-indent them all

File: ntemplates/delivery/detail.html (hunk 8 above)
Comment (this-username-is-taken) [inline ntemplates/delivery/detail.html:72]: The reason I removed those is that the detail page is for delivery guy to pick up grocery from the stores. So for each order, we only need something to identify/distinguish those orders. In this case, a name is enough. I'm removing redundant content to clean up the clutter for better viewing.

When the delivery person delivers, he will go to the invoice page to see the actual address/phone number etc. I will stylize that page later.

File: ntemplates/delivery/detail.html (hunk 7 above)
Comment (this-username-is-taken) [inline ntemplates/delivery/detail.html:42]: I will show `status` in the table below as it's not something crucial for the delivery person while grocery shopping in store.

File: ntemplates/delivery/base.html [hunk 11]
```diff
@@ -1,6 +1,10 @@
 {% extends "common/base.html" %}
 {% load static %}
 
+{% block extra_meta %}
+	<meta name="viewport" content="width=device-width">
```
Comment (gregwym) [inline ntemplates/delivery/base.html:5]: indent with space please.

**Comment by gregwym** [comment]
Lets keep the discussion in comment instead of in-line comments. 

You wrote

> The reason I removed those is that the detail page is for delivery guy to pick up grocery from the stores. So for each order, we only need something to identify/distinguish those orders. In this case, a name is enough. I'm removing redundant content to clean up the clutter for better viewing.
> When the delivery person delivers, he will go to the invoice page to see the actual address/phone number etc. I will stylize that page later.

I disagree. The detail page is used for the entire delivery process. The idea behind this is, if Kevin lost his smartphone, he can still print this page on paper and do the delivery. You can re-organize this page, but we must keep everything a delivery man needs on the same page. 

The invoice page is not designed for delivery guy to use. It is for the customer. Eventually we are going to add account system and we may not even allow a delivery guy to access that page anymore. That is also why the invoice page is under `order` app, but we are dealing with the `delivery` app.

File: ntemplates/delivery/detail.html [hunk 12]
```diff
@@ -1,19 +1,45 @@
...
 {% load tz %}
 
 {% localtime on %}
 
-{% block content %}
 
-<h1>Delivery Window</h1>
-<h2>{{ delivery_window.store }}: {{ delivery_window }}</h2>
+{% block extra_script %}
+	<script type="text/javascript" src="{% static "javascript/delivery.js" %}"></script>
```
Comment (gregwym) [inline ntemplates/delivery/detail.html:12]: Same here, use 2-space. keep it the same across the page.

File: static/javascript/delivery.js [hunk 13]
```diff
@@ -0,0 +1,5 @@
+$(document).ready(function() {
+  $('.hide_button').click(function(){
+    $(this).parent().hide();
+  });
+});
```
Comment (gregwym) [inline static/javascript/delivery.js:5]: I don't quite like this hide idea, cuz once you hide, there is no way to un-hide it unless you refresh. Most of the time, people need to double check. So giving a simple checkbox besides the item in shopping list is enough.

**Comment by gregwym** [comment]
I also want to explain it bit more on the reason of having two sections and what they are indent to do. 

The details page are divided in to sections, and each section represent one step of the delivery process. 
- Shopping list  
  As its name, it should contain everything a delivery guy need to know during shopping. 
- Order list
  It is for delivery guy to pack up orders into different bags. (And was intended for route planing as well.)

I like the address section you added, it can be a dedicated section for delivery, but having only address will not do the job. We must consider carefully, what are the information a delivery guy need in order to complete the step.

**Comment by gregwym** [comment]
Another thing I wanna point out related to PR. Keep one PR focusing on solving one single issue. If its delivery page, then just work on the delivery page. Use different branch & PR for different issue.
